sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineScaleSetImpl.java

```diff
@@ -1709,26 +1695,61 @@ protected void afterCreating() {
             .getByResourceGroupAsync(this.resourceGroupName(), this.name());
     }
 
+
     // Helpers
     //
+    private void adjustProfileForFlexibleMode() {
+        if (this.orchestrationMode() == OrchestrationMode.FLEXIBLE) {
+            if (this.innerModel().virtualMachineProfile().networkProfile().networkInterfaceConfigurations() != null) {
+                this.innerModel().virtualMachineProfile().networkProfile().networkInterfaceConfigurations().forEach(virtualMachineScaleSetNetworkConfiguration -> {
+                    if (virtualMachineScaleSetNetworkConfiguration.ipConfigurations() != null) {
+                        virtualMachineScaleSetNetworkConfiguration.ipConfigurations().forEach(virtualMachineScaleSetIpConfiguration -> {
+                            // this property is not allowed to appear when creating vmss in flexible mode, though it's defined in the swagger file
+//                            if (CoreUtils.isNullOrEmpty(virtualMachineScaleSetIpConfiguration.loadBalancerInboundNatPools())) {
+                            virtualMachineScaleSetIpConfiguration.withLoadBalancerInboundNatPools(null);
+//                            }
+                        });
+                    }
+                });
+            }
+            this.innerModel()
+                // upgradePolicy is not supported in flexible vmss
+                .withUpgradePolicy(null)
+                .virtualMachineProfile().networkProfile()
+                // NetworkApiVersion must be specified when creating in flexible mode
+                .withNetworkApiVersion(NetworkApiVersion.TWO_ZERO_TWO_ZERO_ONE_ONE_ZERO_ONE);
+        }
```

Comment:
Does it always has `virtualMachineProfile() != null`?

Reply:
Yes, in the current flow, it does.